### PR TITLE
Slight clarification of stack exhaustion

### PR DESCRIPTION
**Pull request by jfbastien**

### CODE REVIEW

File: Nondeterminism.md [hunk 1]
```diff
@@ -45,7 +45,9 @@ currently admits nondeterminism:
      may vary between platforms.
  * Environment-dependent resource limits may be exhausted. A few examples:
    - Memory allocation may fail.
-   - Program stack may get exhausted.
+   - Program stack may get exhausted (e.g. because function call depth is too big,
+     or functions have too many locals, or infinite recursion). Note that this stack
+     isn't located in the program-accesible linear memory.
```
Comment (kripken) [inline Nondeterminism.md]: "accessible" has 2 "s"s.

File: Nondeterminism.md [hunk 2]
```diff
@@ -45,7 +45,9 @@ currently admits nondeterminism:
      may vary between platforms.
  * Environment-dependent resource limits may be exhausted. A few examples:
    - Memory allocation may fail.
-   - Program stack may get exhausted.
+   - Program stack may get exhausted (e.g. because function call depth is too big,
```
Comment (kripken) [inline Nondeterminism.md]: Most but not all style guides suggest a comma after "e.g."

**Comment by kripken** [comment]
lgtm

**Comment by jfbastien** [comment]
Seems uncontended, merging based on @kripken's lgtm.

**Pull request merged**